doc/manual/src/glossary.md

```diff
@@ -11,6 +11,28 @@
     The location in the file system where store objects live. Typically
     `/nix/store`.
 
+    From   the  perspective   of   the  location   where  Nix   is
+    invoked, the  Nix store can be  referred to
+    as a "_local_" or a "_remote_" one:
+
+    + A *local  store* exists  on the local  filesystem of
+      the machine where Nix is  invoked. You can use other
+      local stores  by passing  the `--store` flag  to the
+      `nix` command.
+
+    + A  *remote store*  exists  anywhere  other than  the
+      local  filesystem. One  example is  the `/nix/store`
+      directory on another machine,  accessed via `ssh` or
+      served by the `nix-serve` Perl script.
+
```

Comment:
I would not start with this. We can do TCP to a local store in principle, avoiding use of the file system. We can use NFS and do file system access to storage that is far away. "local vs remote" is a misnomer from unjustified assumptions about how the file system is implemented.

Even in the implemented, the daemon on ones machine is communicated to with a `RemoteStore` subclass even though file system access is also used! (Should be called `RpcStore` or something instead.)

Reply:
> We can use NFS and do file system access to storage that is far away.

Okay; I have changed "local filesystem" to "filesystem".

> We can do TCP to a local store in principle, avoiding use of the file system.

Yes but you can't do builds in such a store.

Can you suggest a different term than "local" to express this distinction?

Note: the term "local store" is already in use to express this distinction:

https://github.com/NixOS/nix/blob/4823067247471aebdc24dceb21e13ee3d5411cb8/src/nix/run.cc#L44

Reply:
Bu you *can* do builds over TCP store, as long as the store "on the other end" can mount the items in some filesystem. That's how remote building works!

I'm with you that it would nice to break up the `Store` class further, but figuring out how to do this is surprisingly tricky.

Reply:
> Bu you can do builds over TCP store, as long as the store "on the other end" can mount the items in some filesystem.  That's how remote building works!

Come on.

You can *ask* a remote builder to do builds for you, in its local store on a filesystem that it has mounted.  And then you can ask it to send the build results back to you.

I agree it would be wonderful if we could erase any assumptions about things being on a local filesystem, and all uses of "local store" that currently exist in the source code and documentation.  Whenever that happens we should definitely delete this glossary entry.